ocaml/xapi/xapi_pool.ml

```diff
@@ -1077,6 +1077,75 @@ let designate_new_master ~__context ~host =
     Xapi_pool_transition.attempt_two_phase_commit_of_new_master ~__context true peers my_address
   end
 
+let management_reconfigure ~__context ~network =
+  (* Disallow if HA is enabled *)
+  let pool = Helpers.get_pool ~__context in
+  if Db.Pool.get_ha_enabled ~__context ~self:pool then
+    raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
+
+  (* Disallow if network provided is Host internal management network *)
+  let network_other_config = Db.Network.get_other_config ~__context ~self:network in
+  if (List.mem_assoc Xapi_globs.is_host_internal_management_network network_other_config) &&
+    (List.assoc Xapi_globs.is_host_internal_management_network network_other_config = "true") then
```

Comment:
Unfortunately not. There is an helper hidden somewhere in one of xapi modules, but it essentially does `try List.assoc key map = "true" with _ -> false`.